grunnbelop/src/main/kotlin/no/nav/dagpenger/grunnbelop/Grunnbelop.kt

```diff
@@ -6,50 +6,128 @@ import java.time.LocalDate
+        ),
+        Regel.Minsteinntekt to Gyldighetsperiode(
+            fom = LocalDate.of(2018, Month.JUNE, 4)
+        )
+    ),
+    Grunnbeløp.FastsattI2017 to mapOf(
+        Regel.Grunnlag to Gyldighetsperiode(
+            fom = LocalDate.of(2017, Month.MAY, 1)
+        ),
+        Regel.Minsteinntekt to Gyldighetsperiode(
+            fom = LocalDate.of(2017, Month.MAY, 29)
+        )
+    ),
+    Grunnbeløp.FastsattI2016 to mapOf(
+        Regel.Grunnlag to Gyldighetsperiode(
+            fom = LocalDate.of(2016, Month.MAY, 1)
+        ),
+        Regel.Minsteinntekt to Gyldighetsperiode(
+            fom = LocalDate.of(2016, Month.MAY, 30)
+        )
+    ),
+    Grunnbeløp.FastsattI2015 to mapOf(
+        Regel.Grunnlag to Gyldighetsperiode(
+            fom = LocalDate.of(2015, Month.MAY, 1)
+        ),
+        Regel.Minsteinntekt to Gyldighetsperiode(
+            fom = LocalDate.of(2015, Month.JUNE, 1)
+        )
+    )
 )
 
-fun getGrunnbeløpForMåned(dato: YearMonth): Grunnbeløp {
+private val grunnbeløp = gyldighetsperioder.flatMap { (grunnbeløp, mappings) ->
+    mappings.map { (regel, mapping) ->
+        GrunnbeløpMapping(
+            fom = mapping.fom,
+            grunnbeløp = grunnbeløp,
+            regel = regel,
+            iverksattFom = grunnbeløp.iverksattFom
+        )
+    }
+}.toSet().sortedByDescending { it.fom }
+
+@Deprecated(
+    message = "Du må angi regel først, så måned",
+    replaceWith = ReplaceWith("getGrunnbeløpForRegel(regel).forMåned(dato)")
+)
+fun getGrunnbeløpForMåned(dato: YearMonth): GrunnbeløpMapping {
     return grunnbeløp.first { it.gjelderFor(LocalDate.of(dato.year, dato.month, 10)) }
 }
 
-fun getGrunnbeløpForDato(dato: LocalDate): Grunnbeløp {
+@Deprecated(
+    message = "Du må angi regel først, så dato",
+    replaceWith = ReplaceWith("getGrunnbeløpForRegel(regel).forDato(dato)")
+)
+fun getGrunnbeløpForDato(dato: LocalDate): GrunnbeløpMapping {
     return grunnbeløp.first { it.gjelderFor(dato) }
 }
 
-fun Grunnbeløp.gjelderFor(dato: LocalDate): Boolean {
+fun getGrunnbeløpForRegel(regel: Regel): Set<GrunnbeløpMapping> {
+    return grunnbeløp.filter { it.gjelderFor(regel) }.toSet()
+}
+
+fun Set<GrunnbeløpMapping>.forDato(dato: LocalDate): GrunnbeløpMapping {
+    return utenFramtidigeGrunnbeløp()
+        .first { it.gjelderFor(dato) }
+}
+
+fun Set<GrunnbeløpMapping>.forMåned(dato: YearMonth): GrunnbeløpMapping {
+    return this.first { it.gjelderFor(LocalDate.of(dato.year, dato.month, 10)) }
+}
+
+private fun Set<GrunnbeløpMapping>.utenFramtidigeGrunnbeløp(): List<GrunnbeløpMapping> {
+    val dato = LocalDate.now()
+    return this.filter { it.iverksattFom.isBefore(dato).or(it.iverksattFom.isEqual(dato)) }
+}
+
+fun GrunnbeløpMapping.gjelderFor(dato: LocalDate): Boolean {
     return !(dato.isBefore(this.fom))
-}
\ No newline at end of file
+}
+
+fun GrunnbeløpMapping.gjelderFor(regel: Regel): Boolean {
+    return this.regel == regel
+}
+
+data class GrunnbeløpMapping(
```

Comment:
nitpicking: likte ikke helt navnet - eller egentlig `..Mapping`- har vi noe annet ord på det?

Reply:
`FastsattGrunnbeløp` ?

Reply:
ah, etter siste commit gir det mening :)

Reply:
Jeg kommer ikke på noe bedre. Det er jo et objekt som sier noe om hvilket Grunnbeløp er gyldig til hvilken tid for hvilken regel.

`GrunnbeløpGyldighetsRegel`? Det blir litt rart å bruke `regel` der. `GrunnbeløpValidityRule`?

Reply:
Det er jo en slag `strategy`. Kanskje `GrunnbeløpPolicy`?

Reply:
`GrunnbeløpPolicy` klinger bra, bedre en mapping synes jeg